File: 3.js
// The prime factors of 13195 are 5, 7, 13 and 29.
//
// What is the largest prime factor of the number 600851475143 ?

function primeFact(num){
  var numArr = [];
  // var numCheckArr = [];

  for (x = 2; x <= num; x++){
    while (num % x === 0){
      numArr.push(x);
      num = num / x;
    }
  }
  return numArr[numArr.length-1];
}

primeFact(600851475143);


// function primeFact(num){
//   var numArr = [];
//   var numCheckArr = [];
//   //evenly divisible
//   // if (num % x === 0 )
//   for (x = 1; x <= num; x++){
//     if (num % x === 0){
//       numArr.push(x);
//     }
//   }
//   // at this point x is the number we are trying to determine if prime
//   // range of 1 through x
//   console.log(numArr);
//   for (var x in numArr){
//     for (i = 1; i <= numArr[x]; i++){
//       if (numArr[x] % i === 0){
//         console.log(numArr[x]);
//       }
//     }
//   }
//   // numArr.pop();
//   console.log(numArr);
// }
//
// primeFact(13195);

File: python_solutions/3.py
#The prime factors of 13195 are 5, 7, 13 and 29.

#What is the largest prime factor of the number 600851475143 ?

def prime_finder(num):
    num_list = []

    for possible_factor in range(2,num):
        if num % possible_factor == 0:
            num_list.append(possible_factor)

    print num_list

    

    for factor in num_list:
        if factor %  3 == 0:
            num_list.remove(factor)

    print num_list

prime_finder(13195)
